src/standalone/Vegetation/Canopy.jl

```diff
@@ -444,36 +444,36 @@ function ClimaLSM.make_update_aux(
         n_leaf = hydraulics.n_leaf
         PlantHydraulics.lai_consistency_check.(n_stem, n_leaf, area_index)
         (; retention_model, conductivity_model, S_s, ν) = hydraulics.parameters
-        @inbounds @. ψ[1] = PlantHydraulics.water_retention_curve(
+        @inbounds @. ψ.:1 = PlantHydraulics.water_retention_curve(
             retention_model,
-            PlantHydraulics.effective_saturation(ν, ϑ_l[1]),
+            PlantHydraulics.effective_saturation(ν, ϑ_l.:1),
             ν,
             S_s,
         )
 
         @inbounds for i in 1:(n_stem + n_leaf - 1)
-
-            @. ψ[i + 1] = PlantHydraulics.water_retention_curve(
+            ip1 = i + 1
+            @. ψ.:($$ip1) = PlantHydraulics.water_retention_curve(
```

Comment:
I think it should be okay to use `:($$(i + 1))` here, and it might be more clear. But this way with `ip1` is fine too

Reply:
thanks! this seems to work. I had tried this:
```
julia> @inbounds for i in 1:(n_stem + n_leaf - 1)
           @. ψ.:($$i+1) = FT(0)
       end
ERROR: MethodError: no method matching getproperty(::ClimaCore.Fields.Field{ClimaCore.DataLayouts.DataF{NTuple{11, Float64}, Vector{Float64}}, ClimaCore.Spaces.PointSpace{ClimaCore.DataLayouts.DataF{ClimaCore.Geometry.LocalGeometry{(3,), ClimaCore.Geometry.ZPoint{Float64}, Float64, StaticArraysCore.SMatrix{1, 1, Float64, 1}}, Vector{Float64}}}}, ::Expr)
```

but not with the additional parantheses

Reply:
Hm - this doesnt work after all, 
```
julia> @inbounds for i in 1:(n_stem + n_leaf - 1)
             @. ψ.:($$(i+1)) = FT(0)
       end

julia> @inbounds for i in 1:(n_stem + n_leaf - 1)
             @. ψ.:($$(i+1)) = PlantHydraulics.water_retention_curve(
               retention_model,
               PlantHydraulics.effective_saturation(ν, ϑ_l.:($$(i+1))),
               ν,
               S_s,
           )
       end
ERROR: syntax: invalid syntax "ϑ_l.((i + 1))" around REPL[96]:2
Stacktrace:
 [1] top-level scope
   @ REPL[96]:1
```